app/src/components/Post/Post.tsx

```diff
@@ -0,0 +1,51 @@
+import { ChevronDown, ChevronRight } from "lucide-react";
+import React, { useState } from "react";
+import './Post.scss';
+
+export type Comment = {
+    id: string;
+    userInitials: string;
+    text: string;
+}
+
+export type PostProps = {
+    id: string;
+    body: string;
+    comments: Comment[];
+}
+
+function Post(props: PostProps) {
+    const [isExpanded, setIsExpanded] = useState(false);
```

Comment:
This code functions, but it is fragile. It will break whenever UX introduces an additional method to preview the comments, such as previewing the most popular comments. You will need to broaden the scope of `isExpanded` to encompass the new *most popular comments* scenario. 

The correct approach to this type of modeling is to utilize Typescript's 'union types'. An example would be: `type Preview = 'collapsed' | 'expanded'`. This allows for flexibility in expansion. (Regrettably, developers often default to collapsing every two-state variable into a boolean, which isn't correct).

In conclusion, incorporate a union type for the comments preview and utilize it in place of the `isExpanded`.